From 9011d5d9645b8281ce52203996265371e3c2c816 Mon Sep 17 00:00:00 2001
From: Dhananjay Latkar <quic_dlatkar@quicinc.com>
Date: Tue, 6 Aug 2024 16:10:13 +0530
Subject: [PATCH] netstandby: add wifi priv info in netstandby_entry_info
 struct

netstandby_wifi_priv_info contains linkid information of MLD iface

Change-Id: Iddeae3ac44e4600917a228f71fda60eeccf5843a
Signed-off-by: Dhananjay Latkar <quic_dlatkar@quicinc.com>
---
 include/uapi/linux/netstandby.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/uapi/linux/netstandby.h b/include/uapi/linux/netstandby.h
index 7026a43624f2..1e68d1b65d2f 100644
--- a/include/uapi/linux/netstandby.h
+++ b/include/uapi/linux/netstandby.h
@@ -62,6 +62,15 @@ struct netstandby_nss_priv_info {
 	uint16_t flags;				/**< NSS valid flags */
 };
 
+/*
+ * netstandby_wifi_priv_info
+ *	WiFi private information
+ */
+struct netstandby_wifi_priv_info {
+	int link_id[MAX_INTERFACE]; 		/**< Link ids of wakeup interfaces */
+	int link_id_cnt;			/**< Link ids of wakeup interfaces */
+};
+
 /*
  * netstandby_event_compl_info
  */
@@ -93,6 +102,7 @@ struct netstandby_entry_info {
 	struct net_device *dev[MAX_INTERFACE]; 		/**< List of designated interfaces to exit network standby mode */
 	struct netstandby_exit_trigger_rule tuple;	/**< Trigger rule informataion */
 	struct netstandby_nss_priv_info nss_info;	/**< NSS specific entry information */
+	struct netstandby_wifi_priv_info wifi_info;	/**< WiFi specific entry information */
 	int iface_cnt;					/**< Number of designated wakeup interface */
 };
 
-- 
2.34.1

